lib/imagemounter/index.js

```diff
@@ -0,0 +1,119 @@
+import { readFile, createReadStream } from 'fs';
+import { BaseServicePlist } from '../base-service';
+import { fs } from '@appium/support';
+import B from 'bluebird';
+import log from '../logger';
+const MOBILE_IMAGE_MOUNTER_SERVICE_NAME = 'com.apple.mobile.mobile_image_mounter';
+const { exists, lstat } = fs;
+class ImageMounter extends BaseServicePlist {
+
+    _checkReturnError(ret) {
+        if (ret.Error) {
+            throw new Error(ret.Error);
+        }
+    }
+
+    /**
+     * Lookup for mounted images.
+     * @param {string?} imageType Type of image, `Developer` by default.
+     * @returns {Promise<Buffer[]>} Signature of each mounted image.
+     */
+    async lookup(imageType = 'Developer') {
+        const ret = await this._plistService.sendPlistAndReceive({
+            Command: 'LookupImage',
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        return ret.ImageSignature || [];
+    }
+
+    /**
+     * Check if developer image is mounted.
+     */
+    async isDeveloperImageMounted() {
+        return (await this.lookup()).length > 0;
+    }
+
+    /**
+     * Mount image for device.
+     * @param {string} imageFilePath The file path of image.
+     * @param {string} imageSignatureFilePath The signature file path of given `DeveloperDiskImage.dmg`
+     * @param {string?} imageType Type of image, `Developer` by default.
+     */
+    async mount(imageFilePath, imageSignatureFilePath, imageType = 'Developer') {
+        //check file stats
+        if (!await exists(imageFilePath)) {
+            throw new Error(`The image file provided not exists: ${imageFilePath}`);
+        }
+        const imageFileStat = await lstat(imageFilePath);
+        if (imageFileStat.isDirectory()) {
+            throw new Error(`The given image file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
```

Comment:
This is redundant; instead, try/catch the `lstat` call, since `lstat` will reject if it doesn't exist. You can check the code prop of the `NodeJS.ErrnoException` rejection reason to determine why; `'ENOENT'` will be "doesn't exist". Please repeat below for `imageSignatureFilePath`.

```suggestion
        /** @type {import('node:fs').Stats|undefined} */
        let imageFileStat;
        try {
          imageFileStat = await lstat(imageFilePath);
        } catch (err) {
          if (/** @type {NodeJS.ErrnoException} */(error).code === 'ENOENT') {
            throw new Error(`The provided image file doesn't exist: ${imageFilePath}`);
          }
          throw err;
        }
        if (imageFileStat?.isDirectory()) {
            throw new Error(`The provided path was expected to be a file, but it is a directory: ${imageFilePath}`);
        }
```